specifyweb/frontend/js_src/lib/components/Merging/index.tsx

```diff
@@ -361,6 +381,58 @@ function MergeButton<SCHEMA extends AnySchema>({
   );
 }
 
+export function Status({
+  mergingId,
+  loadingBar: [_, setLoadingBar],
+}: {
+  readonly mergingId: string;
+  readonly loadingBar: GetSet<boolean>;
+}): JSX.Element {
+  const [status, setStatus] = React.useState<string>();
+  const [total, setTotal] = React.useState<number>();
+  const [current, setCurrent] = React.useState<number>();
+
+  React.useEffect(() => {
+    const fetchStatus = () =>
+      void ajax(`/api/merging_status/${mergingId}/`, {
+        // eslint-disable-next-line @typescript-eslint/naming-convention
+        headers: { Accept: 'application/json' },
+      })
+        .then((data) => {
+          console.log(data);
+          if (data === null) return undefined;
+          else {
+            // setStatus(data.status);
+            // setTotal(data.total);
+            // setCurrent(data.current);
+            globalThis.setTimeout(fetchStatus, 2000);
+          }
+          return undefined;
+        })
+        .catch(softFail);
+    fetchStatus();
+  }, [mergingId]);
+
+  return (
+    <Dialog
+      buttons={
+        <Button.Danger onClick={() => setLoadingBar(false)}>
```

Comment:
when the button is clicked, a network request should be made notifying the back-end that user wants to abort, right?

Reply:
yes but it doesn't exist yet...